tools/generator/Java.Interop.Tools.Generator.ObjectModel/Method.cs

```diff
@@ -105,7 +105,7 @@ internal string CalculateEventName (Func<string, bool> checkNameDuplicate)
 
 		public string ConnectorName => $"Get{Name}{IDSignature}Handler";
 
-		public string EscapedCallbackName => $"cb_{JavaName}{IDSignature}";
+		public string EscapedCallbackName => $"cb_{JavaName}{IDSignature}".Replace ("-", "_x45_").Replace ("$", "_x36_");
 
 		public string EscapedIdName => "id_" + JavaName.Replace ("<", "_x60_").Replace (">", "_x62_") + IDSignature;
```

Comment:
I see in this instance you only replace the JavaName and not the signature. Does it matter?

Reply:
I'm not sure I understand?

Reply:
This line is `JavaName.Replace ("<", "_x60_").Replace (">", "_x62_")`
and then new one is `"cb_{JavaName}{IDSignature}".Replace (...`

Reply:
I wonder if this is relevant: https://github.com/xamarin/java.interop/pull/498#discussion_r332102772

Have a "single" "ensure the string is a valid C# identifier" method and use that consistently ~everywhere.

Reply:
I'll be merging this PR, but on the whole I believe @mattleibow is correct, and `EscapedIdName` should likewise use `IdentifierValidator.CreateValidIdentifier()` as well.

However, AFAICT `EscapedIdName` is only used for the legacy `JNIEnv`-based output format, not the newer `XAJavaInterop1` type, so I'm not entirely concerned about this.  We should *prefer* `XAJavaInterop`.